modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java

```diff
@@ -1384,6 +1384,10 @@ private ObjectField _updateObjectField(
 				newObjectField.getReadOnly(), readOnlyConditionExpression));
 
 		if (objectDefinition.isApproved()) {
+			if (newObjectField.isSystem()) {
+				newObjectField.setRequired(required);
+			}
+
```

Comment:
@devpedrotavares please confirm with the team the expected result here.

I think we should **only** allow changing from required to not required after the definition is published.

Please see this [comment](https://liferay.slack.com/archives/C032VGYA2UD/p1706110600628849?thread_ts=1706097138.712499&cid=C032VGYA2UD).

Reply:
Thank you. I'll wait for a definition there